system/src/system_cloud_internal.cpp

```diff
@@ -1064,23 +1090,21 @@ int Spark_Handshake(bool presence_announce)
         if (presence_announce) {
             Multicast_Presence_Announcement();
         }
-        LOG(INFO,"Send subscriptions");
-        spark_protocol_send_subscriptions(sp);
-        // important this comes at the end since it requires a response from the cloud.
         spark_protocol_send_time_request(sp);
-        Spark_Process_Events();
```

Comment:
@avtolstoy I couldn't find out why it might be important to send this at the end of the application handshake and why we run a loop iteration afterwards, so any thoughts are welcomed. So far it looks as a safe change.

Reply:
Isn't this call here just to make sure that the comms event loop is immediately run? I don't see any special reason why we should call it immediately though.

Reply:
ah, found it: https://github.com/particle-iot/device-os/commit/2cde8bc02d08d5f4d74dcb5cc1ec10ef8c6da6f4